tests/rpm-ostree/main.yml

```diff
@@ -0,0 +1,565 @@
+
+    - name: Commit new local branch
+      command: ostree commit -b local-branch --tree=ref=local-branch --add-metadata-string version=test
+      register: new_commit
+
+    - name: Upgrade
+      command: rpm-ostree upgrade --install {{ g_pkg }}
+      register: ros_upgrade
+      until: ros_upgrade.rc == 0
+      retries: 5
+      delay: 5
+
+    # reboot
+    - include: ../../common/ans_reboot.yml
+
+    # verify upgrade info
+    - include: roles/rpm_ostree_status_verify/tasks/main.yml
+      num_deployments: 2
+      deployment: 0
+      expected:
+        booted: true
+        base-checksum: "{{ new_commit.stdout }}"
+
+    # verify installation of {{ g_pkg }}
+    - include: roles/rpm_ostree_install_verify/tasks/main.yml
+      vars:
+        package: "{{ g_pkg }}"
+
+    - name: Rebase back to original deployment
+      command: rpm-ostree rebase {{ refspec }} {{ head_csum }}
+
+    # reboot
+    - include: ../../common/ans_reboot.yml
+
+    # verify rebase info
+    - include: roles/rpm_ostree_status_verify/tasks/main.yml
+      num_deployments: 2
+      deployment: 0
+      expected:
+        booted: true
+        base-checksum: "{{ head_csum }}"
+
+    # verify package is still layered
+    - include: roles/rpm_ostree_install_verify/tasks/main.yml
+      vars:
+        package: "{{ g_pkg }}"
+
+    # refresh rpm-ostree status variables
+    - include: roles/rpm_ostree_status/tasks/main.yml
+
+    - name: Pull last two commits
+      command: ostree pull --commit-metadata-only --depth=1 {{ refspec }}
+      register: osp
+      until: osp.rc == 0
+      retries: 5
+      delay: 5
+
+    - name: Get HEAD-1 checksum
+      command: ostree rev-parse {{ refspec }}^
+      register: hmo_csum
+
+    - name: Deploy checksum {{ hmo_csum.stdout }}
+      command: rpm-ostree deploy {{ hmo_csum.stdout }} --uninstall {{ g_pkg }}
+      register: ros_deploy
+      until: ros_deploy.rc == 0
+      retries: 5
+      delay: 5
+
+    # reboot
+    - include: ../../common/ans_reboot.yml
+
+    # verify package is no longer installed
+    - include: roles/rpm_ostree_uninstall_verify/tasks/main.yml
+      vars:
+        package: "{{ g_pkg }}"
+
+    # verify deploy with uninstallation info
+    - include: roles/rpm_ostree_status_verify/tasks/main.yml
+      num_deployments: 2
+      deployment: 0
+      expected:
+        booted: true
+        checksum: "{{ hmo_csum.stdout }}"
+
+    # rpm-ostree upgrade
+    - include_role:
+        name: rpm_ostree_upgrade
+
+    # reboot
+    - include: ../../common/ans_reboot.yml
+
+    # verify package is still not installed
+    - include: roles/rpm_ostree_uninstall_verify/tasks/main.yml
+      vars:
+        package: "{{ g_pkg }}"
+
+    # verify upgrade info
+    - include: roles/rpm_ostree_status_verify/tasks/main.yml
+      num_deployments: 2
+      deployment: 0
+      expected:
+        booted: true
+        checksum: "{{ head_csum }}"
+
+    - name: Cleanup
+      command: rpm-ostree cleanup -rpmb
+
```

Comment:
Another instance to verify that the cleanup was successful with `rpm_ostree_status_verify`?